lib/Folder/FolderManager.php

```diff
@@ -286,11 +286,21 @@ private function getGroups(int $id): array {
 	/**
 	 * @throws Exception
 	 */
-	public function canManageACL(int $folderId, string $userId): bool {
+	public function canManageACL(int $folderId, IUser $user): bool {
+		$userId = $user->getUId();
 		if ($this->groupManager->isAdmin($userId)) {
 			return true;
 		}
 
+		// Call private server api
```

Comment:
Check for what this is needed and add comment

Reply:
Basically if this returns false the user can't manage the ACL in the admin UI or in the folder sidebar. Added a bit of doc